unison-runtime/src/Unison/Runtime/Machine.hs

```diff
@@ -877,6 +879,13 @@ dataBranch mrf stk (Test1 u cu df) = \case
   DataG _ t seg
     | maskTags t == u -> (cu,) <$> dumpSeg stk seg S
     | otherwise -> pure (df, stk)
+  Foreign f
+    | Just m <- maybeUnwrapForeign Rf.hmapRef f -> case m of
```

Comment:
Seems like it could be beneficial to add pattern synonyms for each of the types we rewrite so we can benefit from the pattern-match-coverage checker wherever they're used.